### ISSUE TITLE
<input type=range> does not coincide with the beginning and end due to padding

### ISSUE THREAD

**Issue by bsr203** [twbs/bootstrap#19812]
Hello,

please see the example http://codepen.io/anon/pen/EKeqYv

the range slider is started at 0, but because of padding it looks like it can't select extreme values (0 and 1). I think the slider should coincide with the start and end in this case. It works fine if I remove padding (padding: .5rem .75rem;) and border.

thanks.
bsr.

**Comment by twbs-lmvtfy** [comment]
Hi @bsr203!

You appear to have posted a live example (http://codepen.io/anon/pen/EKeqYv.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 20, column 5: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by RyanZim** [comment]
@bsr203 What browser(s) and OS have you tested with?

**Comment by bsr203** [comment]
only chrome

Version 52.0.2718.0 canary (64-bit)
Version 50.0.2661.86 (64-bit)

**Comment by bsr203** [comment]
well, as per above lint comment I may not use `form-control` with range. But, then the behavior is not consistent ike no 100% width etc. Why not range support `form-control` ?

**Comment by RyanZim** [comment]
@bsr203 I'm not sure if lmvtfy is correct there; it's often wrong about v4.

This seems to affect WebKit browsers; here are cross-browser screenshots:

- Chrome ![chrome](https://cloud.githubusercontent.com/assets/17342435/14854541/2099c9f2-0c5e-11e6-879a-f970b209dd76.JPG)

- Safari ![safari](https://cloud.githubusercontent.com/assets/17342435/14854564/2f532c0e-0c5e-11e6-80bb-88da7046dbd4.JPG)

- Firefox ![firefox](https://cloud.githubusercontent.com/assets/17342435/14854573/379eb8ec-0c5e-11e6-9d5d-559cba9fb171.JPG)

- IE ![ie](https://cloud.githubusercontent.com/assets/17342435/14854578/3e11009a-0c5e-11e6-807c-50fa7aaf1009.JPG)

**Comment by wolfy1339** [comment]
lmvtfy is correct here as per v4 docs, remove the form-control class and repost an updated codepen.
Check out my PR #19468, I added a custom range class that styles the range input and it works well.

**Comment by bsr203** [comment]
here is the updated pen http://codepen.io/anon/pen/VaEazv

the padding issue is gone, but form elements are not consistent. I may be able to make it similar to other inputs like 100% width padding and block behavior through css, but may better if it is supported through existing class. Will close if you all think otherwise.

thanks again for confirming the behavior.

**Comment by wolfy1339** [comment]
Add this class to it, `.form-control-range`, it should restore functionality as it was in v3 (100% width and display block)

**Comment by bsr203** [comment]
thanks. I didn't know about that class.

http://codepen.io/anon/pen/KzGzrx

no 100% width, but has block behavior. 
https://github.com/twbs/bootstrap/blob/v4-dev/dist/css/bootstrap.css#L2156

you think that's acceptable

**Comment by wolfy1339** [comment]
Didn't know about that. 
For now, use the latest version in the `v4-dev` branch and add the `.w-100` class to make it 100% width like you want it.

**Comment by mdo** [comment]
I don't see a way around this without mucking around with potentially gnarly some browser hack stuff. This is just how ranges in Webkit look (with just the `.form-control-range` class). Punting as a won't fix for now.
